### PR TITLE
miniusAreas' big brig overhaul

### PR DESCRIPTION
**Pull request by ghost**
:cl: miniusAreas
add: Majorly remapped security
/:cl:

sec1
sec2
sec3

### CODE REVIEW

**Comment by ghost** [comment]
I need to fix the areas, because currently it's using areas with a VAREDITED NAME VAR.

**Comment by miniusAreas** [comment]
I was unable to upload this to a PR because Git threw a hissyfit. I gave Steamport the mapfile, and is allowed to manage the PR for me.

**Comment by Wilchenx** [comment]
I feel like you're overbuffing security quite a bit with this

**Comment by alexkar598** [comment]
Agreed

**Comment by Keluandrie** [comment]
Bikes are cool, but also broken as shit, as they're basically a downsideless jetpack that let's you block stuff as any other vehicle does. The bay would be better fit for nerfed bikes or a pod system, neither of which are in place

**Comment by Cactus2388** [comment]
Where does the transit tube begin? Pretty iffy about genpop being able to be "visited" like that. One emag and all the prisoners are out.

>miniegun instead of full egun
>energy shotgun (the combat shotty thing) is way too strong for roundstart
>regular shotgun instead of the riot shotguns
>The mix of reinforced glass and plasma glass is really ugly

>the showers and meeting room are literally the same room
>siren helmet needs to be removed
>no recharger in the officer room

>space bike bay exists
>turrets randomly surrounding it outside
>it has disablers for some reason?

Big no from me dog

**Comment by Ziiro** [comment]
The space bikes are a ridiculous and unnecessary buff. The meeting room with the showers in it is totally wack. Big fuck no to the turrets. Even bigger fuck no to giving security **even more guns.**

Also the sec protolathe probably should be in the locker room.

**Comment by miniusAreas** [comment]
Sigh, here we go explaining.

>The armory starts bolted and locked down to prevent those pesky caps and such from rushing the armory. Any warden looting the armory roundstart needs a jobban @Ziiro @Cactus2388 

>Those are minings Perimiter Defence Turrets. All they shoot at is hostile fauna, like carp, that could take a bike driver by surprise

>The bikes require a sec key to function, and only work in space. Its planned to become pods but EVERYONE IS MAKING A FUSS ABOUT A PLACEHOLDER

>Theres like, 1-2 more guns than normal armory. Half of them are the mini-eguns, and disablers.

>These are the pump version of the shotgun, meaning they function the same as a riot

>The bikes are there to give sec a fighting chance against antags in hardsuits with jetpack upgrades. Their hardsuits dont have this, and mean that you need both a jetpack ON YOUR BACK SLOT, and a AIR TANK IN YOUR SUIT STORAGE, leaving you with no room to put a gun. You also cannot pull whilest on a bike.

>Those are cameras, @Ziiro. I can see where you mistook them

> @Jeff-jefferson-jeffery The tube to the brig is protected fully by rwalls and a shutter at the Customs area, and is shut by default. SoP will require visitors to comply to a bag and pocket search, and to have any items that could aid in a escape to be removed. The officer is to monitor the visitor till they return, and close the shutters.

>Minieguns are to acomadate a weaker gun so that we could have the eshotguns.

> @Cactus2388 The protolathe is in the wardens quarters to prevent every single sec officer printing a AEG once science researches it, allowing for the warden to have full confidence his officers dont have a fucking railgun on them.

> @Jeff-jefferson-jeffery It takes 700 hits or so to break a reinforced windoor. Im pretty sure they may not take any damage from it, either. Besides, it leads to a room they wont be able to access a way out from.

@Wilchenx @Keluandrie @alexkar598 



Sorry to any of you who were worried that sec was becoming a dictatorshipial powerhouse. If anyone is actualy looting the armory roundstart, then somethings wrong, the armory is alert color-coded for a fucking reason. Meta Armory has 16 guns, while this has 18+. many of that are minieguns or non-lethal options.

Chasing someone into space as sec has been nigh - impossible without bikes. It literaly ends up with the officer floating in space, the officer dead, or the officer loosing the criminal. To even have sec use the bikes, it requires the warden to open BOTH EVA and the bay doors, meaning the bikes wont be used for non-space non-sec activites. There is a plan to give the crew and other departments bikes as well.

The next department getting a special treatment is science. Expect a lot of cool stuff, however im not sure how to make this and it work without having this merged. I allready passed the armory pass @quantum, and they seemed mostly ok with it.

If someones rushing the red/blue armory roundstart, ahelp it. Just because we cant make everyone perfect doesnt mean sec cant have a few things to even the playing field in some realy fucking unfair places. If someones riding a spacebike indoors, they have a warning upon shiftclicking them NOT TO RIDE THEM AROUND INDOORS. It permits a godly smiting upon them, so stand back and watch the fun i guess.

Again, this was to even the playing field against some realy unfair circumstances, not to make sec uberstonkmurdervalidbonerhunters. Science, medbay, engineering. and service are getting some similar upgrades, making them even with sec, and even adding new ways for antags to do their thing. I just need the PR merged to start...... for some reason....

**Comment by miniusAreas** [comment]
The helm of justice is just a fun little cool thing for sec to mess with and get yelled at for it. If its actualy used in some way that isnt realy fair or some shit, ill remove it.

**Comment by Ziiro** [comment]
>Theres like, 1-2 more guns than normal armory. Half of them are the mini-eguns, and disablers.

Adding guns is still adding guns. Don't add guns.

>The bikes are there to give sec a fighting chance against antags in hardsuits with jetpack upgrades. Their hardsuits dont have this, and mean that you need both a jetpack ON YOUR BACK SLOT, and a AIR TANK IN YOUR SUIT STORAGE, leaving you with no room to put a gun. You also cannot pull whilest on a bike.

It's almost like sec not having good space movement options and having to select their loadout carefully when going to engage in space combat is a  design choice or something and fighting in space is inherently very dangerous

You could leave the bay, but nix the provided bikes and make it so they have to order them from cargo.

**Comment by miniusAreas** [comment]
@Ziiro The bikes are varedits, and i know 0 code. They are placeholders, for now. Yeesh.

Also, its legit disablers. If you get taken down by disabler fire, somethings wrong.

**Comment by ghost** [comment]
I don't understand why people don't like this, unless you don't like the spacebikes or extra guns.

I removed all but of one of the turrets (they only shoot carp anyways, like the mining base turrets)

**Comment by ghost** [comment]
And as for people worried about visitation: it's pretty secure. Only sec can open the windoors, plus the plasma glass has locked sec barriers and electrified grilles under them.

Also I added a shutter to the visitation.

**Comment by Cactus2388** [comment]
Still iffy about visitation. It just takes one emaggy boi and people can sneak out, and shutters can easily be broken. I really like the idea of a visitation center and would like to see it playtested but I do not like the rest of the rework. It still leaves me the question: Where does the transit tube begin?

**Comment by Wilchenx** [comment]
>It just takes one emaggy boi and people can sneak out

emergent gameplay

**Comment by LeRuffleFace** [comment]
Leaving the prisoner and the visitor unattended would be a pretty bonehead move in the first place.

**Comment by ghost** [comment]
@Cactus2388 Transit tube begins near arrivals garden

@LeRuffleFace is right... PLUS THE AI CAN WATCH THEM, BECAUSE THE VISITATION CAMERA TRIGGERS A DAMN MOTION ALARM!

**Comment by Cactus2388** [comment]
His office also gets a Medical Department console? I don't think that's intended.

**Comment by Wilchenx** [comment]
there's a .357 autorevolver and power fist in HoS's locker, his pet is replaced for no reason, there is a unaccessible filing cabinet in his office.

The whole office was for some reason also moved from security to occupy a part of gateway.

**Comment by ghost** [comment]
Good news: no more revolver or power fist for HoS. Sorry buddy, but you aren't thanos.

**Comment by Cactus2388** [comment]
Alright here we go:
Genpop
>holy shit it's ugly
>it's glowing red for some reason? Probably not intentional
>The second basketball hoop is ugly and you only really need one
>transit tube is long as fuck and is really ugly

EVA/Space bike

>Space bikes
>Really ugly floor lights don't respect layers I guess? They're popping up above everything else, including machinery, also glowing red.
>Buttons inside the Space bike bay are broken, don't do anything when clicked
>no way to keep all the air venting from the space bike bay when the doors open
>disablers are unnecessary
>Security EVA has way too many EVA suits
>The space carp turret is very hug boxxy

Firing range/Security meeting/Showers
>Showers have no showering heads
>**FIRING RANGE WASNT LISTED IN THE PR PLEASE ADD IT**
>Firing range has MMIs in it?
>Shit tons of security barricades?
>I'd rather all the chairs in the room be the same than have half of them matching, half not
>remove the helmet of justice that's a cargo thing
>/obj/item/robot_module/security if I'm mistaken this is the Security borg module?

Armory
>private security force having a grenade launcher
>regular shotgun, not riot
>clashing plasmaglass windows and non plasmaglass windows
>3 scatter laser guns
>2 temp guns
>2 energy shotguns
>5 mini eguns
>3 advtasers
>3 disablers, not including the ones in the space bike bay
>Tons of powerpacks spawned above the energy guns, none of them are compatible for any of the guns in the armory

Head of Security Office
>**NOT LISTED IN THE PR PLEASE ADD IT**
>Has a varedited pet specifically made for the HOS named Nekopatta
>Nekopatta has a type in it's description
>Has a medical department computer, likely no intentional.
>Consoles block the cabinet
>Head of Security's power fist is unnecessary and spawns above the locker, not in it.
>Head of Security spawns with a .357 Unica Autorevolver, which is unnecessary and OP
>Office is in a public hall and holds a traitor objective, traitor item, and a .357 revolver

**Comment by ghost** [comment]
@Cactus2388 
Red glow is fixed.
Scatter laser guns, energy shotguns, and non-riot shotguns are gone now. Along with HoS's powerfist and revolver.
Gonna get minius to move the HoS's office back to sec. 
Grenade launcher isn't OP, it only launches was sec has: flashbangs and tear gas.
Security borgs are enabled on the server, I believe.
I might replace the transit tube with a quantum pad.
Will fix up storage, or get minius to.

**Comment by ghost** [comment]
placeholders are bad and not an excuse to add things until someone else replaces them. speedbikes suck massive dick

**Comment by ghost** [comment]
@miniusAreas and tg doesnt require any new maps. we've been trying to get rid of maps as is.

File: _maps/map_files/NSS-Sunset/NSS-Sunset.dmm [hunk 1]
```diff
@@ -28414,6 +28414,7 @@
 /area/security/checkpoint/customs)
 "beT" = (
 /obj/machinery/quantumpad{
+	circuit = null;
```
Comment (ghost) [inline _maps/map_files/NSS-Sunset/NSS-Sunset.dmm]: holy shit what are you doing
go back to transit pipes this is not okay

File: _maps/map_files/NSS-Sunset/NSS-Sunset.dmm [hunk 2]
```diff
@@ -84189,6 +84190,7 @@
 	icon_state = "4-8"
 	},
 /obj/machinery/quantumpad{
+	circuit = null;
```
Comment (ghost) [inline _maps/map_files/NSS-Sunset/NSS-Sunset.dmm]: yamero

**Review by ghost — CHANGES_REQUESTED** [review]

File: _maps/map_files/NSS-Sunset/NSS-Sunset.dmm (hunk 1 above)
Comment (ghost) [reply _maps/map_files/NSS-Sunset/NSS-Sunset.dmm]: no, transit pipes are ugly

**Comment by Ziiro** [comment]
dreamseeker_2018-09-19_19-06-47

remove whatever the fuck this is supposed to be and put the HoS office back in here. And add back his spider, what the fuck is this cake cat?

File: _maps/map_files/NSS-Sunset/NSS-Sunset.dmm (hunk 1 above)
Comment (ghost) [reply _maps/map_files/NSS-Sunset/NSS-Sunset.dmm]: you had to remove a possible advantage smart players could get by snowflaking it in the map file do you realize how much of a bad design decision this is

**Comment by miniusAreas** [comment]
@Ziiro Im fixing shit right now. Araneius isnt a thing every station has, its mappers preference, for instance, on Delta, the HoS has a magicarp.

**Comment by Ziiro** [comment]
Also run the tube from the recreation area. It'll be shorter, intersect with less, and make more sense. Not to mention dropping them out near security so they can respond in a decent timeframe if shit is going awry on it.

**Comment by azzzertyy** [comment]
Varedit the visitation shutters to make them 21 armor, theyre only 20 i believe and thats just enough to be damaged by thrown spears. Also, please change the mirror room, i feel like you could just do one small bathroom like the captains office somewhere near that area to make it look a lot better, And replace the flooring in it for white tiles. Also, i know your reason for the storage room looking bad was that it was a storage room, but, this defeats the point and i dont see why any area needs to look bad.

**Comment by azzzertyy** [comment]
Also, i dont get the hate for the cak. It isnt something thats gonna be abused, it just provides a small healing buff to anyone in security whenever the eat it or give donuts to it. I doubt any user is going to use it for this reason, and just provides a fun in-game reason as to why to have it. But, i think itd be easy to compromise here and just add the spider somewhere else in security.

**Comment by Wilchenx** [comment]
virgin cak vs chad spider

**Comment by Ziiro** [comment]
don't

Don't thumbs up your own PR, it's bad form.

**Review by alexkar598 — DISMISSED** [review]
massive var edit on objects/structures on maps is not a good thing to do and honestly the whole remap is shit and unneeded but thats not what this review is for.

**Comment by ghost** [comment]
@alexkar598 WTF are you talking about? There's no fucking reason to add new types for a varedit that only occurs once or twice. Or for pixel_x/y changes. Or name changes.

**Comment by ghost** [comment]
lol you know damn well that's a different field entirely because it's purely visual and doesn't affect people as much

**Comment by ghost** [comment]
Pics update

**Comment by Ziiro** [comment]
Much better. Although I'm unsure how I feel about the sec lathe being in the Warden's office

**Comment by ghost** [comment]
prevents sec officers from shitting out tesla revolvers and zapping the nearest greytide with em

**Comment by miniusAreas** [comment]
@Qustinnus Yeah, sorry if your gonna shit all over me, but plasma windows or reinforced glass with firelocks?

**Comment by miniusAreas** [comment]
@steamp0rt Please remove from testmerge until the pipes are fixed. Also, send the .dmm so i can work on it at afterschool shit and pull a BS office for @alexkar598 while im at it.

**Comment by miniusAreas** [comment]
Should also mention @SunsetStationuser while im at it, sparky remove from testmerge, theres some fatal errors.

**Comment by ghost** [comment]
Closed in favor of https://github.com/SunsetStation/SunsetStation/pull/282

**Pull request closed without merging**